pkg/steps/template.go

```diff
@@ -588,6 +588,9 @@ func waitForPodCompletionOrTimeout(ctx context.Context, podClient coreclientset.
 		if event.Type == watch.Deleted {
 			return false, appendLogToError(fmt.Errorf("the pod %s/%s was deleted without completing after %s (failed containers: %s)", pod.Namespace, pod.Name, podDuration(pod).Truncate(time.Second), strings.Join(failedContainerNames(pod), ", ")), podMessages(pod))
 		}
+		if !isPodRunning(pod) && time.Since(pod.CreationTimestamp.Time) > 15*time.Minute {
+			return false, fmt.Errorf("pod didn't start running within 15 minutes: %s", getReasonsForUnreadyContainers(pod))
+		}
```

Comment:
Some types of failures will never resolve themselves no matter how long we wait.  We could fail immediately here if we have enough information to identify those.

Reply:
So you are suggesting that we shouldn't use a timeout at all and instead fail on specific conditions, like ImagePullBackoff?

Reply:
I think a hard timeout is still useful, and we cannot avoid having one because some errors (some types of `ImagePullBackoff`, for example) are transient.

If we have enough information in the pod status to determine if the error is transient or not, it would be nice to fail immediately.  If not, this is fine as it is.

Reply:
But do we have any kind of error here that is guaranteed not to be transient?

Reply:
IIRC, the registry acting up results in a transient `ImagePullBackoff`, for example.

Reply:
Yeah so that in turn means there is basically no error for which we can say for sure that its terminal, right?

Reply:
That was kind of my question: does the pod information we have at that point allow us to determine whether it is terminal or not?